Explorer/Assets/Scripts/ECS/StreamableLoading/Cache/RefCountStreamableCacheBase.cs

```diff
@@ -72,16 +74,33 @@ public void Unload(IPerformanceBudget frameTimeBudget, int maxUnloadAmount)
 
             for (int i = listedCache.Count - 1; frameTimeBudget.TrySpendBudget() && i >= 0 && maxUnloadAmount > 0; i--)
             {
-                (TLoadingIntention key, TAssetData asset) = listedCache[i];
-                if (!asset.CanBeDisposed()) continue;
+                if (TryUnloadAsset(i))
+                    maxUnloadAmount--;
+            }
 
-                asset.Dispose();
-                cache.Remove(key);
-                listedCache.RemoveAt(i);
+            inCacheCount.Value = cache.Count;
+        }
 
-                maxUnloadAmount--;
-            }
+        // Returns true if the asset was unloaded
+        private bool TryUnloadAsset(int i)
+        {
+            (var key, var asset) = listedCache[i];
+            if (!asset.CanBeDisposed()) return false;
+
+            asset.Dispose();
+            cache.Remove(key);
+            listedCache.RemoveAt(i);
+            return true;
+        }
+
+
+        public void UnloadImmediate()
```

Comment:
I think branching between `Unload` and `UnloadImmediate` is not necessary, it just creates a new set of methods to maintain:

- logic is duplicated
- if you supply `int maxUnloadAmount = int.Max` and `NullPerformanceBudget` it will cover your needs without necessity to maintain a second way

Reply:
Done!